repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/files/FileHelperTest.java

```diff
@@ -0,0 +1,62 @@
+package fr.inria.spirals.repairnator.process.files;
+
+import static org.junit.Assert.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Test;
+
+import fr.inria.spirals.repairnator.process.files.FileHelper;
+import fr.inria.spirals.repairnator.process.inspectors.JobStatus;
+import fr.inria.spirals.repairnator.process.inspectors.ProjectInspector;
+import fr.inria.spirals.repairnator.process.step.CloneRepository;
+import fr.inria.spirals.repairnator.process.utils4tests.ProjectInspectorMocker;
+
+
+public class FileHelperTest {
+	
+	private File tmpDir;
+	
+	@After
+	public void tearDown() throws IOException {
+	    FileHelper.deleteFile(tmpDir);
+	}
+	
+	@Test
+	public void testRemoveNotificationFromTravisYML() throws IOException{
+	    File resourceFile = new File("./src/test/resources/travis-file/.travis.yml");
+	    tmpDir = Files.createTempDirectory("test_removeNotificationFromTravisYML").toFile();
+	    FileUtils.copyFileToDirectory(resourceFile, tmpDir);
+	
+	    JobStatus jobStatus = new JobStatus(tmpDir.getAbsolutePath());
+	    ProjectInspector inspector = ProjectInspectorMocker.mockProjectInspector(jobStatus, tmpDir.getAbsolutePath());
+	    CloneRepository cloneStep = new CloneRepository(inspector);
+	
+	    FileHelper fileHelper = new FileHelper();
+	    fileHelper.removeNotificationFromTravisYML(tmpDir, cloneStep);
+	
+	    File bak = new File(tmpDir.getAbsolutePath()+"/bak.travis.yml");
+	    File travis = new File(tmpDir.getAbsolutePath()+"/.travis.yml");
+	
+	    assertTrue(bak.exists());
+	    assertTrue(travis.exists());
+	
+	    boolean detected = false;
+	    List<String> lines = Files.readAllLines(travis.toPath());
+	    for(String l: lines){
+		if(l.contains("notification")){
+		    assertTrue(l.trim().startsWith("#"));
```

Comment:
I'm not sure about this test: here it seems you only check that the line notification is commented. However the notification contains a block, so how do you ensure that you won't have something like:

```yml
#notifications:
   emails: 
     x.y@z.org
```

which would be wrong?

Reply:
In fact, this is my fault, because I passed this code snippet to Benjamin. I copied this code snippet from the existing integration test.
I will think on how to fix that and I will tell Benjamin.

Reply:
Could we look at all consecutive lines that begin with ```#``` and see that the next line after these should not begin with a tab/space? As I've understood it, ```notifications:``` are the *lowest level* and thus we need to make sure that everything before the next lowest level is commented. This assumes that the indentation is correct however, which I am unsure if we can rely on.

Reply:
I looked up yml-files and they seem to be very strict about indentation, so made a test as mentioned above checking for spaces in the beginning of lines.

Reply:
Your strategy seems good for me. It's already better than the strategy on the old integration test.